config.md

```diff
@@ -265,6 +265,27 @@ Annotations are key-value maps.
 }
 ```
 
+## HypervisorPath
```

Comment:
I think it depends where the line of demarcation lies- who/what is the ultimate arbiter in the OCI world?

In the current docker world, my understanding is that docker itself will "own" the OCI config file (but it would be great to get clarification on docker/docker#21324). Currently, docker is hard-coded to launch containerd (well, "docker-containerd"). But in that world, containerd gets to decide which OCI runtime it will launch ("containerd --runtime runc").  And if that runtime were, say, runv rather than runc, then runv gets to decide which actual hypervisor is ultimately launched to host the workload.

Clearly, this is but one example, but is this the way we want the system to work generally? Or should the OCI config specify:

1) which runtime to use ("HypervisorPath" here, but should probably be renamed "RuntimePath").

2) which hypervisor that runtime should use (optional).

If the spec shouldn't specify this, then yes the field can be dropped.

But if these values should be encoded in the config, docker could continue to call containerd, but containerd would then simply consume the config.json and launch the appropriate runtime.  Assuming that docker owns config.json and that file is generated by docker, docker can perform path checks, etc when passed the appropriate configuration data (again, see docker/docker#21324) which it will use to generate config.json. And docker would know which runtimes were available on the system since the admin would have provided the details to docker.

Currently, the OCI bundle includes config.json, but does that imply the file is "read-only"? Or will "OCI managers" (is there an official term?) like docker be able to supplement (in other words rewrite) this configuration based on locally available resources? If so, what are they allowed to change in the file?  Can they only supplement, or are they permitted to change anything and everything?

Looking at [application bundle builders](/opencontainers/specs/blob/master/README.md#application-bundle-builders), we see:

    > Because the configuration includes host-specific settings, application
    > bundle directories copied between two hosts may require configuration
    > adjustments.

This implies that config.json can be modified to atleast add supplementary data.  Further, some of the config specified by config-linux.md almost guarantees the file will need to be modified before passing to the runtime (memory, CPU shares, cgroups, etc).

To me, this implies it might be better to have a "static" (effectively read-only) config file that contains the minimum possible config for the bundle (config.json). That file would be owned not by the "OCI manager", but by the bundle author.  Hence, it would be illegal for the OCI manager to, say, change process.args.

Additionally, the bundle could also provide an optional separate runtime.json that could specify reasonable defaults for certain settings. However, that runtime.json could be modifiable by the OCI manager in any way it deems appropriate (the file is owned by the OCI manager). Regarding runtime.json, yes, I know we've already been there :-)